### PR TITLE
CASMCMS-8974: Add additional information about using multi-tenancy with BOS via CURL

### PR DESCRIPTION
**Pull request by jsl-hpe**
# Description
This adds additional information for administrators who desire to interact with the BOS API with the curl command.

Relates to:
- CASMCMS-8974

# Checklis
- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

### CODE REVIEW

File: operations/boot_orchestration/Multi_tenancy_with_BOS.md [hunk 1]
```diff
@@ -21,15 +21,35 @@ resources specific to that tenant.
...
+select the tenant that they are operating under, as it is possible for one user to be a part of multiple tenants. In the
+case of `BOS`, even if you are part of multiple individual tenant groups, only the resources that are allocated to a
+specific provided tenant, as referenced by name, are affected.
+
+1. (`ncn-mw#`) Administrators may provide tenant information via an HTTP verb in the form of a header to affect resources
+that are part of a specific `TAPMS` tenant.
+
+    ```bash
+    curl -H "Cray-Tenant-Name: red" -H "Authorization: Bearer ${TOKEN}" -H 'Content-Type: application/json' https://api-gw-service-nmn.local/apis/bos/v2/components/
+    ```
```
Comment (nrockershousen) [inline operations/boot_orchestration/Multi_tenancy_with_BOS.md]:
````suggestion
Administrators may provide tenant information via an HTTP verb in the form of a header to affect resources
that are part of a specific `TAPMS` tenant.

(`ncn-mw#`) 

```bash
curl -H "Cray-Tenant-Name: red" -H "Authorization: Bearer ${TOKEN}" -H 'Content-Type: application/json' https://api-gw-service-nmn.local/apis/bos/v2/components/
```

````

**Review by rustydb — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Review by nrockershousen — APPROVED** [review]

File: operations/boot_orchestration/Multi_tenancy_with_BOS.md [hunk 2]
```diff
@@ -21,15 +21,38 @@ resources specific to that tenant.
...
+all sessions and session templates across all tenants. However, system administrators will not be able to create, retrieve,
 update, or delete specific records that belong to a tenant without authenticating to that specific tenant. This is
 because the resources are name-spaced to avoid naming collisions between tenants. The name-spacing means that there can
-be two or more records with the same name, and BOS does not know which resource is requested unless the tenant is
+be two or more records with the same name, and `BOS` does not know which resource is requested unless the tenant is
 specified. To access tenant owned resources, the system administrator will need to masquerade as that tenant.
+
+## BOS API Access with Tenancy
+
+`BOS` v2 API introspects request information pertaining to tenant information, if provided. If you have tenant information
```
Comment (jsollom-hpe) [inline operations/boot_orchestration/Multi_tenancy_with_BOS.md]: What does this sentence mean?
`BOS` v2 API introspects request information pertaining to tenant information, if provided.

I think you're saying the following which I would suggest could be a replacement for that sentence.
The `BOS` v2 API identifies the tenant based on information passed in the request. A tenant must always pass information identifying itself -- their tenant ID -- when making a request.

I think the additional following re-wording is also more clear and could replace the paragraphs up to the `Note:`.

The CrayCLI will supply the tenant ID as part of any requests made through it, provided that the tenant logged into CrayCLI via 'cray init'.

Alternately, when making requests to the Cray API directly, the user must supply the tenant ID as part of the request.

The tenant ID is provided as a header value in the request. It is of the form `Cray-Tenant-Name: <tenant name>`.

File: operations/boot_orchestration/Multi_tenancy_with_BOS.md [hunk 3]
```diff
@@ -21,15 +21,38 @@ resources specific to that tenant.
...
+populated by the CrayCLI, it will be passed into the associated `BOS` command. Otherwise, if you are interacting with the
+API directly and are scripting against it, you may provide this information by hand, via curl.
+
+Note: `OPA` prevents unauthorized tenant requests from accessing the `BOS` API, so the value you choose for 
+`Cray-Tenant-Name: ` must match your user provided access token within your request. If your user access token is not
+part of the tenant name you are issuing, your command will fail. This allows users who are part of multiple tenancies to
+select the tenant that they are operating under, as it is possible for one user to be a part of multiple tenants. In the
+case of `BOS`, even if you are part of multiple individual tenant groups, only the resources that are allocated to a
+specific provided tenant, as referenced by name, are affected.
```
Comment (jsollom-hpe) [inline operations/boot_orchestration/Multi_tenancy_with_BOS.md]: specific --> specifically

File: operations/boot_orchestration/Multi_tenancy_with_BOS.md [hunk 4]
```diff
@@ -21,15 +21,38 @@ resources specific to that tenant.
...
+be two or more records with the same name, and `BOS` does not know which resource is requested unless the tenant is
 specified. To access tenant owned resources, the system administrator will need to masquerade as that tenant.
+
+## BOS API Access with Tenancy
+
+`BOS` v2 API introspects request information pertaining to tenant information, if provided. If you have tenant information
+populated by the CrayCLI, it will be passed into the associated `BOS` command. Otherwise, if you are interacting with the
+API directly and are scripting against it, you may provide this information by hand, via curl.
+
+Note: `OPA` prevents unauthorized tenant requests from accessing the `BOS` API, so the value you choose for
```
Comment (jsollom-hpe) [inline operations/boot_orchestration/Multi_tenancy_with_BOS.md]: Rewording: HPE written `OPA` rules prevent unauthorized...

File: operations/boot_orchestration/Multi_tenancy_with_BOS.md [hunk 5]
```diff
@@ -21,15 +21,38 @@ resources specific to that tenant.
...
+API directly and are scripting against it, you may provide this information by hand, via curl.
+
+Note: `OPA` prevents unauthorized tenant requests from accessing the `BOS` API, so the value you choose for 
+`Cray-Tenant-Name: ` must match your user provided access token within your request. If your user access token is not
+part of the tenant name you are issuing, your command will fail. This allows users who are part of multiple tenancies to
+select the tenant that they are operating under, as it is possible for one user to be a part of multiple tenants. In the
+case of `BOS`, even if you are part of multiple individual tenant groups, only the resources that are allocated to a
+specific provided tenant, as referenced by name, are affected.
+
+Administrators may provide tenant information via an HTTP verb in the form of a header to affect resources
```
Comment (jsollom-hpe) [inline operations/boot_orchestration/Multi_tenancy_with_BOS.md:52]: Reword: Administrators may provide this tenant ID in the header of a request to 'masquerade' as a tenant and affect that tenant's resources. The `OPA` rules allow this.

File: operations/boot_orchestration/Multi_tenancy_with_BOS.md [hunk 6]
```diff
@@ -21,15 +21,38 @@ resources specific to that tenant.
...
 specified. To access tenant owned resources, the system administrator will need to masquerade as that tenant.
+
+## BOS API Access with Tenancy
+
+`BOS` v2 API introspects request information pertaining to tenant information, if provided. If you have tenant information
+populated by the CrayCLI, it will be passed into the associated `BOS` command. Otherwise, if you are interacting with the
+API directly and are scripting against it, you may provide this information by hand, via curl.
+
+Note: `OPA` prevents unauthorized tenant requests from accessing the `BOS` API, so the value you choose for 
+`Cray-Tenant-Name: ` must match your user provided access token within your request. If your user access token is not
```
Comment (jsollom-hpe) [inline operations/boot_orchestration/Multi_tenancy_with_BOS.md]: This user-provide access token is the bearer token indicated in the example as
Authorization: Bearer ${TOKEN}
correct?

**Review by jsollom-hpe — CHANGES_REQUESTED** [review]

File: operations/boot_orchestration/Multi_tenancy_with_BOS.md (hunk 6 above)
Comment (jsl-hpe) [reply operations/boot_orchestration/Multi_tenancy_with_BOS.md]: Yes.

**Pull request merged**